### DISCUSSION TITLE
how to import class/functions defined in emitter-framework

### DISCUSSION THREAD

**Discussion by hyang691** [microsoft/typespec#3596] [Q&A] [answered]
I referenced document https://typespec.io/docs/extending-typespec/emitter-framework and try to create a demo emitter.
I want to add import for class CodeTypeEmitter  and function createAssetEmitter.  I tried two ways and both get error when running  command **tsp compile api.tsp**.  Anyone have idea on how to import them? 

 **"@typespec/compiler": "0.55.0"**

**Way 1:**
import { CodeTypeEmitter} from '@typespec/compiler/dist/src/emitter-framework/type-emitter';
import { createAssetEmitter} from '@typespec/compiler/dist/src/emitter-framework/asset-emitter';
**Error for Way 1:**
Package subpath './dist/src/emitter-framework/type-emitter' is not defined by "exports" in /home/ubuntu/analytics-sdk/sdks/emitter/node_modules/@typespec/compiler/package.json  

Way 2:
import { CodeTypeEmitter, createAssetEmitter} from '@typespec/compiler/dist/src/emitter-framework';

**Error for way 2:**
 Package subpath './dist/src/emitter-framework/type-emitter' is not defined by "exports" in /home/ubuntu/analytics-sdk/sdks/emitter/node_modules/@typespec/compiler/package.json

**Comment by bterlson** [comment]
Very sorry I missed this discussion post!

I think you need to just import `@typespec/compiler/emitter-framework`. Let me know if that works.

Also it's ridiculous that we don't document this import! We'll fix that.

**Comment by hyang691** [comment]
Thanks @bterlson, I tested and it does not work in my environment.

![image](https://github.com/microsoft/typespec/assets/173017702/9bc71358-6c5a-42c2-aba8-6d0c3c0681b1)

**Reply by timotheeguerin** [reply, marked as answer]
Assuming this error is from TypeScript, is your package configured for `node16` module resolution? https://typespec.io/docs/next/extending-typespec/basics#d-install-and-initialize-typescript

**Reply by hyang691** [reply]
Thanks @timotheeguerin, it works now.
